### DISCUSSION TITLE
Softconsole targeting the MPF500T

### DISCUSSION THREAD

**Discussion by rockoguido** [Mi-V-Soft-RISC-V/miv-rv32-documentation#9] [Bare Metal Embedded Software]
Do we have any examples of the use of SoftConsole targeting the MPF500T?     All of our eval boards are MPF300T based.   What's driving this  I have a user using the SE200 baseboard from Sundance (https://www.sundance.technology/som-cariers/pci-express-fmc-carriers/solar-express-200/)  along with the SOM1 module from Sundance  (https://www.sundancedsp.com/som/polarfire/som-pf1-fpga-module-with-mpf500-1fcg1152/)

A Libero based model seems to able to be loaded into the fabric on the SOM1 but when trying to load an application via SoftConsole we appear to connect to the target (no errors asserted) but we don't get to a known state, i.e. halt at Main.   

The JTAG logic on the SE200 is a direct copy of the JTAG logic on the PolarFire Eval Board, down to the FTDI chip being used (FT4232HQ) and the pin outs.   Of course we can connect to the the PolarFire Eval board via SoftConsole and debug.   
The memory map is not the issue, the only difference being the use of the MPF500T on the Sundance SOM1 vs the MPF300T on the PolarFire Eval card from Microchip.

**Comment by CLappin** [comment]
Hi @rockoguido.

A couple of questions to aid debugging.
1. Which processor are you using on the board?
2. Is your design passing timing?
3. Are you using LSRAM or DDR. If DDR, are they the same DDR modules on both the Eval board and Sundance board?

If the memory maps we're the same I wouldn't expect this to be a SoftConsole issue, more of an issue in the Libero design.
The fact that it isn't halting at main would suggest that the memory (if DDR) isn't set up correctly, could you try with an LSRAM to see if we can rule out SoftConsole being the issue?

Thanks,
Ciaran

**Reply by rockoguido** [reply]
Hi Ciaran,

Processor is the MPF500T, see https://www.sundancedsp.com/som/polarfire/som-pf1-fpga-module-with-mpf500-1fcg1152/

I can't give you a clear answer yet on timing and for sure we will look at the DDR modules being used on the Eval board and the Sundance board.     The good news is that I will have hardware in hand next week, It will be a combination of the SOM module I mention above and the SE200 baseboard.    see https://www.sundance.technology/som-cariers/pci-express-fmc-carriers/solar-express-200/

Thanks!
Dave

**Reply by CLappin** [reply]
Hi Dave,

Apologies as I think I may have confused you a little.
Which processor are you targeting from SoftConsole, for example RISC-V or Cortex-M3.
The MPF500T is only an FPGA fabric, there is no hardened RISC-V processor as on the [PolarFire SoC MPFS460T](https://www.microchip.com/en-us/products/fpgas-and-plds/system-on-chip-fpgas/polarfire-soc-fpgas)

If it is the MPF500T you will need to create a design with a fabric RISC-V core in Libero SoC (MIV RV32 is available from the Libero SoC catalog). Then you could you SoftConsole to debug the RISC-V core.  There are also examples available for the [MPF300T devices](https://mi-v-soft-risc-v.github.io/PolarFire-Eval-Kit/) , these would need to be retargeted to the MPF500T device, for example the GPIO and UART pins may differ between the boards that you are targeting. 


Thanks, 
Ciaran

**Reply by rockoguido** [reply]
Hi Ciaran,

Thank you for the clarification.  Yes I am aware that the MFP500T is fabric only by default.    We are loading a design via Libero that includes a MiV along with some basic peripherals, i.e. GPIO and a UART.     As practice I have done the same for the MPF300T via the Microchip Eval board.   That scenario works fine and with the correct design for the MPF300T and the miv-rv32i-systick-blinky example I am able to load this example via SoftConsole and execute the binary and see the UART and GPIO I/O.  
So yes, it was important for me to get used to the work flow using the MFP300T Eval board.

My next steps.   
- in 2 days I will receive the board and SOM module from Sundance and I'll be able to test this on my own.   
- We have  a design built for the 500T board that is similar to the design for the 300T with different I/O pin assignments too.   
- I am open to suggestions and I will let you know how this exercise goes.

Thx again
Dave

**Comment by rockoguido** [comment]
hi, an update here.  I have the following hardware..
https://www.sundance.technology/som-cariers/pci-express-fmc-carriers/solar-express-200/  and SOM1 module
https://www.sundancedsp.com/som/polarfire/som-pf1-fpga-module-with-mpf500-1fcg1152/

The SE200 has the same JTAG logic as Microchip's PolarFire Eval Kit,    a key chage is an update to the program in the EEPROM on the Sundance SE200 so that it matches the program in the same EEPROM on the PolarFire Eval board.  That EEPROM type is the FT4232HQ-REEL.    This change allowed the SRAM on the MPF500T to be accessed correctly

A utility called FT_Prog allowed me to update the EEPROM
At this point am able to load and execute the miv-rv32i-systick-blinky example on this SOM1 (MPF500T) without an issue.   Of course there is a model resident in the fabric that supports the MFP500T along with the use of an MiV, GPIO and UART.    

As a side note, have the UART running at 460800 baud without issue.   The max for this UART is 921600 but have yet to succeed to get that to work and most likely will have issues reliably achieving given the current implementation of character based I/O,  no buffering of data at the TX side and no flow control , however once I setup the UART for Fractional Precision for Baud Rate Generation I may have a chance.

Using the CoreUARTapb

**Reply by CLappin** [reply]
Hi @rockoguido ,

Great to hear that you're making progress on the MPF500 device!
Hopefully everything from here on out goes easier. 
Let us know if you have anymore questions.

Thanks,
Ciaran
